### ISSUE TITLE
Tab completion under pew?

### ISSUE THREAD

**Issue by ColMcp** [berdario/pew#150]
macOS 10.12.6   Python 3.6.2   pew 0.1.26

When I run the CPython REPL from a standard shell tab completion is operative. When I run it in a sub-shell invoked by pew tab completion is not operative.

Is this a bug, or a configuration failure on my part?

**Comment by kennethreitz** [comment]
Confirmed here.

**Comment by kennethreitz** [comment]
the following code activates tab completion:

```python
import rlcompleter, readline
readline.parse_and_bind('tab: complete')
```

**Comment by kennethreitz** [comment]
https://github.com/kennethreitz/pipenv/issues/497

**Comment by ColMcp** [comment]
The above `parse_and_bind()` call didn't work for me but
     `readline.parse_and_bind('bind ^I rl_complete')` did work.

`site.py` from the Python library has the following code in `enablerlcompleter()`

    # Reading the initialization (config) file may not be enough to set a
    # completion key, so we set one first and then read the file.
    readline_doc = getattr(readline, '__doc__', '')
    if readline_doc is not None and 'libedit' in readline_doc:
        readline.parse_and_bind('bind ^I rl_complete')
    else:
        readline.parse_and_bind('tab: complete')

Maybe we have different arrangements for accessing a readline library on our respective machines even though we are both using macOS.

**Comment by berdario** [comment]
This reminds me of https://github.com/berdario/pew/issues/90 :/

Unfortunately, this is another bit of behavior due to virtualenv (and that would be solved by a [rewrite](https://github.com/pypa/virtualenv/pull/697))

(there might be actually another option to prevent this, but it's late and I haven't looked deeply into this)

Apparently the rewrite is not happening :disappointed: . But maybe a minimal rewrite of virtualenv, with just the few features required by us, might be feasible (we wouldn't have [this concern](https://github.com/pypa/virtualenv/pull/697#issuecomment-159084245), since we could invoke it directly as a library, and avoid conflicts/pollution in the PATH)

**Comment by erinxocon** [comment]
@berdario Want to vendorize virtualenv?

**Comment by berdario** [comment]
You mean vendorize and trying to patch this ourselves?

I'd rather try to fix it upstream, and add a version bound on the patch that fixed this would be better

**Comment by erinxocon** [comment]
No, vendorizing just lets us have more control I suppose, but it prevents us from merging any fixes the maintainers put in place.  You way makes sense.

**Comment by nealmcb** [comment]
Is this fixed upstream now? I can't reproduce it.

Using Ubuntu 20.04, Python 3.8.10, Pew 1.2.0, I do 

```python
pew mktmpenv
python
imp<TAB>
```
and it autocompletes to `import `

**Comment by mrx23dot** [comment]
please add tab completion like 
`sudo systemctl s<tab>`
very handy
